src/cmake.ts

```diff
@@ -464,17 +456,24 @@ class ErrorParser extends util.OutputParser {
     private _activeParser: Maybe<DiagnosticParser>;
     private _parserCollection: Set<DiagnosticParser>;
 
-    constructor(binaryDir: string) {
+    constructor(binaryDir: string, parsers: Maybe<string[]>) {

```

Comment:
You are fast! Thanks!

Let me explain changes...

Second thing that is broken with your changes:
a) parseBuildDiagnostics is not taken in account
b) cmake() function originally only handled cmake errors

My latest changes is proposal how this could be handled.
Also this fits to the one (of two) "improvements" I would like to see implemented.
If I use msvc compiler I know that gcc or other parsers will newer have match, so there is no point to execute them. This could be part of "cmake.diagParsers" (ore something similar) setting.
Default value for "cmake.diagParsers" (if not set) would be null/undefined => execute all implemented parsers.